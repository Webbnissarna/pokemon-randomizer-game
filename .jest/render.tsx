import { render as rtlRender } from "@testing-library/react";
import { InitializeColorMode, ThemeProvider } from "theme-ui";

import theme from "../utils/theme";

import "./style.css";

export function render(ui: React.ReactElement) {
  return rtlRender(
    <>
      <InitializeColorMode />
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </>
  );
}
