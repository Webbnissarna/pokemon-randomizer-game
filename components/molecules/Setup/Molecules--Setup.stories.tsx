import { useState } from "react";
import { Flex } from "theme-ui";
import Setup from "./";

export function Default() {
  const [setupOptions, setSetupOptions] = useState({});

  return (
    <Flex sx={{ flexDirection: "column" }}>
      <Setup onPlay={setSetupOptions} />
      <h4>Options:</h4>
      <pre>{JSON.stringify(setupOptions, null, 2)}</pre>
    </Flex>
  );
}
