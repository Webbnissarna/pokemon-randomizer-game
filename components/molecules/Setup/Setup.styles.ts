import { ThemeUIStyleObject } from "theme-ui";

export const mainContainerStyle: ThemeUIStyleObject = {
  flexDirection: "column",
  gap: "md",
};

export const cardBoxStyle: ThemeUIStyleObject = {
  flexDirection: "column",
  gap: "md",
};

export const playerNamesStyle: ThemeUIStyleObject = { gap: "md" };
