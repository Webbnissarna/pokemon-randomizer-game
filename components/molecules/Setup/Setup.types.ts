export interface SetupOptions {
  playerNames: string[];
}

export interface SetupProps {
  onPlay: (setupOptions: SetupOptions) => void;
}

export enum TabValue {
  Solo = "Solo",
  TwoPlayer = "2-player",
}
