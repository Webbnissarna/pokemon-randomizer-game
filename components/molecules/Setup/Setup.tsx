import { ChangeEvent, useMemo, useState } from "react";
import { Flex } from "theme-ui";
import Button from "../../atoms/Button";
import CardBox from "../../atoms/CardBox";
import TabRadioInput from "../../atoms/TabRadioInput";
import TextInput from "../../atoms/TextInput";
import set from "lodash/fp/set";
import { OnChangeEvent } from "../../atoms/TextInput/TextInput.types";
import type { SetupProps, SetupOptions } from "./Setup.types";
import { TabValue } from "./Setup.types";
import {
  cardBoxStyle,
  mainContainerStyle,
  playerNamesStyle,
} from "./Setup.styles";

const tabValues = Object.values(TabValue);

export default function Setup({ onPlay }: SetupProps): JSX.Element {
  const [activeTab, setActiveTab] = useState(TabValue.TwoPlayer);
  const [setupOptions, setSetupOptions] = useState<SetupOptions>({
    playerNames: ["", ""],
  });

  const onTabChange = (event: ChangeEvent<HTMLInputElement>) => {
    setActiveTab(event.target.value as TabValue);
  };

  const playButtonEnabled = useMemo(() => {
    return (
      activeTab === TabValue.Solo ||
      !setupOptions.playerNames.some((value) => value.length === 0)
    );
  }, [activeTab, setupOptions.playerNames]);

  const updatePlayerName = (event: OnChangeEvent, playerNo: number) => {
    setSetupOptions((current) =>
      set(`playerNames[${playerNo}]`, event.target.value, current)
    );
  };

  return (
    <Flex sx={mainContainerStyle}>
      <TabRadioInput
        values={tabValues}
        selected={activeTab}
        onChange={onTabChange}
      />
      <CardBox>
        <Flex sx={cardBoxStyle}>
          {activeTab === TabValue.TwoPlayer && (
            <Flex sx={playerNamesStyle}>
              <TextInput
                placeholder="Player 1 Name"
                value={setupOptions.playerNames[0]}
                onChange={(event) => updatePlayerName(event, 0)}
              />
              <TextInput
                placeholder="Player 2 Name"
                value={setupOptions.playerNames[1]}
                onChange={(event) => updatePlayerName(event, 1)}
              />
            </Flex>
          )}
          <Button
            text="Play"
            variant="neutral"
            size="sm"
            disabled={!playButtonEnabled}
            onClick={() => onPlay(setupOptions)}
          />
        </Flex>
      </CardBox>
    </Flex>
  );
}
