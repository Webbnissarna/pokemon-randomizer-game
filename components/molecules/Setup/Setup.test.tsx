import { render } from "../../../.jest/render";
import Setup from ".";

describe("Setup", () => {
  it("renders", () => {
    const { asFragment } = render(<Setup onPlay={() => undefined} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
