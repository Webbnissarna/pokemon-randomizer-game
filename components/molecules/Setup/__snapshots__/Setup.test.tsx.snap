// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup renders 1`] = `
<DocumentFragment>
  <script>
    (function() { try {
  var mode = localStorage.getItem('theme-ui-color-mode');
  if (!mode) return
  document.documentElement.classList.add('theme-ui-' + mode);
} catch (e) {} })();
  </script>
  <div
    class="css-rcez78-Box"
  >
    <div
      class="css-1qwd1qz-Box"
    >
      <div
        class="css-1d2fn-Box"
      >
        <input
          checked=""
          id=":r0:-2-player"
          name=":r0:"
          type="radio"
          value="2-player"
        />
        <label
          class="css-dzydor-Box"
          for=":r0:-2-player"
        >
          2-player
        </label>
      </div>
    </div>
    <div
      class="css-wimkxn-Box"
    >
      <div
        class="css-1i6ye1s-Box"
      >
        <div
          class="css-p1luab-Box"
        >
          <input
            aria-label="Player 1 Name"
            class="css-kdrqma-Box"
            placeholder="Player 1 Name"
            type="text"
            value=""
          />
          <input
            aria-label="Player 2 Name"
            class="css-kdrqma-Box"
            placeholder="Player 2 Name"
            type="text"
            value=""
          />
        </div>
        <button
          class="css-fy5kkv-Box"
          disabled=""
        >
          Play
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
